tests/AwsLambda/UnitTests/AwsLambdaOpenTracerTests/DataCollectorTests.cs

```diff
@@ -35,32 +35,29 @@ public void DoesWriteDataToCloudWatch()
         [Test]
         public void DoesWriteDataToNamedPipe()
         {
+            if (!Directory.Exists("/tmp"))
```

Comment:
I don't think this test will execute the expected logic. While using the System.IO.Abstractions package is a great idea, but it has MIT license and I don't know what the legal implication might be.

Another approach would be creating a layer of abstraction for `File` so that we can mock it. Below is the the psuedo code:

`
    interface IFileManager
    {
        bool Exists(string path);
        void WriteAllText(string path, string contents);
    }

    public class ConcreteFileManager : IFileManager
    {
        public bool Exists(string path)
        {
            return File.Exists(path);
        }
        public void WriteAllText(string path, string contents)
        {
            File.WriteAllText(path, contents);
        }
    }
` 

`
    internal class LambdaSpanBuilder : ISpanBuilder
    {
        private readonly IFileManager _fileManager;
        ...
        ...
        public LambdaSpanBuilder(string operationName)
        {
            _logger = new Logger();
            _fileManager = new ConcreteFile();
            this._operationName = operationName;
        }
        
        ...
        ...
        public ISpan StartManual()
        {
            ...
            ...
            var rootSpan = new LambdaRootSpan(_operationName, DateTimeOffset.UtcNow, _tags, guid, new DataCollector(_logger, tracer.DebugMode, _fileManager), new TransactionState(), new PrioritySamplingState(), new DistributedTracingState());
            ...
            ...
        }
    }
`
`    internal class DataCollector
    {
        private readonly IFileManager _fileManager;

        ...
        ...

        public DataCollector(ILogger logger, bool debugMode, IFileManager fileManager)
        {
            _debugMode = debugMode;
            _logger = logger;
            _fileManager = fileManager;
        }
    }`

Reply:
Thanks for the help @vuqtran88! I made those changes.